doc/babbage-features.md

```diff
@@ -0,0 +1,35 @@
+# CTL Babbage feature overview
+
+This document is a reference/explainer for the new CTL APIs introduced for Babbage support.
+
+## Reference Inputs
+
+[Reference inputs](https://cips.cardano.org/cips/cip31/#referenceinputs) allow looking at an output without spending it in Plutus scripts.
+
+In CTL reference inputs can be used by providing a value of type `InputWithScriptRef` that specifies whether an output should be spent or referenced. One of the constraints that use it is `mustSpendScriptOutputUsingScriptRef`.
```

Comment:
It's better to mention the `mustReferenceOutput` constraint there, since it allows to simply add a reference input without using it to satisfy script witnessing requirements. Though, we should still leave a note that the reference inputs added using this constraint cannot be used to resolve scripts and also mention `mustSpendScriptOutputUsingScriptRef` and its variants that should be used for this purpose.